apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1686,143 +1657,148 @@ where
                         levels_in_data_path += 1;
                     }
                     let Some(e) = (*e).into() else { continue };
-                    if !was_unsatisfied && !context_map.contains_key(&ctx.named_parameter) {
-                        if let Some(parent_type) = parent_type {
-                            let parent_type = parent_type.parent();
-                            let subgraph_schema =
-                                self.graph.schema_by_source(&ctx.subgraph_name)?;
-                            let mut potential_matches: IndexSet<Name> = Default::default();
-                            ctx.types_with_context_set.iter().for_each(|pos| {
```

Comment:
Bugs:
- The supergraph schema should be used here instead of the subgraph schema.
- The union logic here is inverted, I've added a comment here to better explain it.